### PR TITLE
Add clients for each resource

### PR DESCRIPTION
**Pull request by hawx**

### CODE REVIEW

File: src/lib/client.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import RolesClient from './roles';
+
+export default class {
+  private _client: AppVeyorClient;
+
+  constructor(private _token: string) {
+    this._client = new AppVeyorClient(_token);
+  }
+
+  projects()      { return new ProjectsClient(this._client); }
```
Comment (Codesleuth) [inline src/lib/client.ts]: How about making these property getters?

``` ts
get projects() {
  return this.projectsClient || this.projectsClient == new ProjectsClient(this._client);
}
```

Feels better calling it in dot notation.

File: src/lib/client.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import RolesClient from './roles';
+
+export default class {
+  private _client: AppVeyorClient;
+
+  constructor(private _token: string) {
+    this._client = new AppVeyorClient(_token);
+  }
+
+  get projects()      { return new ProjectsClient(this._client); }
```
Comment (Codesleuth) [inline src/lib/client.ts:17]: You don't wanna make this a singleton?

File: src/lib/appveyor.ts [hunk 3]
```diff
@@ -1,152 +1,74 @@
...
+    const options = {
+      method: method,
+      protocol: this.protocol,
+      port: this.port,
+      hostname: this.hostname,
+      path: path,
+      headers: {
+        'Authorization': `Bearer ${this._token}`,
+        'Content-Type': 'application/json',
+        'Accept': 'application/json'
```
Comment (Codesleuth) [inline src/lib/appveyor.ts:46]: You should be setting `Content-Length`. Best way to handle it is make a `Buffer` first and use `.length` on that:

https://github.com/Codesleuth/esendex-node/blob/master/src/lib/requesthandler.ts#L65

``` js
const dataBuf = Buffer.from(data);
headers['Content-Length'] = dataBuf.length;
```

Comment (Codesleuth) [inline src/lib/appveyor.ts:46]: Although apparently your tests pass??

File: src/lib/client.ts (hunk 2 above)
Comment (hawx) [inline src/lib/client.ts:17]: No

File: src/lib/appveyor.ts (hunk 3 above)
Comment (hawx) [inline src/lib/appveyor.ts:46]: I don't like that this isn't done automatically.

**Pull request merged**